backend/logic/users/users.js

```diff
@@ -167,10 +175,47 @@ var getCompletedCourses = async studentID => {
   return courses;
 };
 
+var getStudentData = async studentID => {
+  let data;
+
+  let conn = await mysql.getNewConnection();
+
+  let courses, major, minor;
+  try {
+    courses = await conn.query(
+      `SELECT course_code, semester
+    FROM course_offerings
+    WHERE (id, semester)
+    IN (SELECT offering_id, semester FROM student_course_offerings WHERE student_id = ?);`,
+      [studentID]
+    );
+    major = await conn.query(
+      `SELECT curriculum_name FROM student_major WHERE student_id = ?;`,
+      [studentID]
+    );
+    minor = await conn.query(
+      `SELECT curriculum_name FROM student_minor WHERE student_id = ?;`,
+      [studentID]
+    );
+    conn.release();
+  } catch (err) {
+    conn.release();
+    console.log(err);
```

Comment:
```suggestion
    console.log(err);
    throw new Error(error);
```
Would this be better?

Reply:
Yes! Since its being caught at the endpoint level